server/segment_rpc.go

```diff
@@ -254,6 +219,78 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 	w.Write(buf)
 }
 
+// Payment receives payment from Gateway and adds it into the orchestrator's balance
+func (h *lphttp) Payment(w http.ResponseWriter, r *http.Request) {
+	payment, segData, oInfo, ctx, err := h.processPaymentAndSegmentHeaders(w, r)
+	if err != nil {
+		return
+	}
+
+	buf, err := proto.Marshal(&net.PaymentResult{Info: oInfo})
+	if err != nil {
+		clog.Errorf(ctx, "Unable to marshal transcode result err=%q", err)
+		return
+	}
+	clog.V(common.DEBUG).Infof(ctx, "Payment processed, current balance = %s", currentBalance(h, payment, segData))
+
+	w.Write(buf)
+}
+
+func currentBalance(h *lphttp, payment net.Payment, segData *core.SegTranscodingMetadata) string {
+	if h == nil || h.node == nil || h.node.Balances == nil || segData == nil || segData.AuthToken == nil {
+		return ""
```

Comment:
If this is only being used in logging, should we do something more descriptive than an empty string?

Reply:
Updated